lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -38,30 +47,53 @@ def self.supported?
         !win32console_enabled?
       end
 
-      def initialize(cmd, debug, pipe_timeout)
+      def self.supported_on_pwsh?
+        # TODO Do we need to check which versions of pwsh are supported on *nix?
+        !win32console_enabled?
+      end
+
+      def initialize(cmd, args = [], options = {})
         @usable = true
 
-        named_pipe_name = "#{SecureRandom.uuid}PuppetPsHost"
+        if Puppet::Util::Platform.windows?
+          # Named pipes under Windows will automatically be mounted in \\.\pipe\...
+          # https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs#L34
+          named_pipe_name = "#{SecureRandom.uuid}PuppetPsHost"
+          # This named pipe path is Windows specific.
+          pipe_path = "\\\\.\\pipe\\#{named_pipe_name}"
+        else
+          # .Net implements named pipes under Linux etc. as Unix Sockets in the filesystem
+          # Paths that are rooted are not munged within C# Core.
+          # https://github.com/dotnet/corefx/blob/94e9d02ad70b2224d012ac4a66eaa1f913ae4f29/src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs#L49-L60
+          # https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Unix.cs#L44
+          # https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Unix.cs#L298-L299
+          named_pipe_name = File.join(Dir.tmpdir, "#{SecureRandom.uuid}PuppetPsHost")
+          pipe_path = named_pipe_name
+        end
+        pipe_timeout = options[:pipe_timeout] || self.class.default_options[:pipe_timeout]
+        debug = options[:debug] || self.class.default_options[:debug]
```

Comment:
Since you've already merged in the defaults, why is this necessary?

Reply:
The initialize method is public so strictly speaking I couldn't guarantee that those options would be passed in.  Should probably mark initialize as private and then I can get rid of that.

Reply:
Derp....You can't make initialize private (It's the constructor)